experimenter/experimenter/experiments/changelog_utils.py

```diff
@@ -87,3 +92,66 @@ def generate_nimbus_changelog(experiment, changed_by, message, changed_on=None):
         message=message,
         changed_on=changed_on,
     )
+
+
+def get_formatted_change_object(field_name, field_diff, changelog, timestamp):
+    event_name = ChangeEventType.GENERAL.name
+    field_instance = NimbusExperiment._meta.get_field(field_name)
+    field_display_name = (
+        field_instance.verbose_name
+        if hasattr(field_instance, "verbose_name")
+        else field_name
+    )
+
+    old_value = field_diff["old_value"]
+    new_value = field_diff["new_value"]
+
+    if isinstance(
+        field_instance,
+        (models.ManyToManyField, models.OneToOneField, models.ManyToOneRel),
+    ):
+        event_name = ChangeEventType.DETAILED.name
+        if field_name in RelationalFields.NATIVE_MODELS:
+            field_model = field_instance.related_model
+            data = field_model.objects.all()
+            values = list(data.filter(pk__in=old_value).values_list("name", flat=True))
+            old_value = values if values else old_value
```

Comment:
What are you trying to do here? Why do we want to fall back to `old_value` if values is an empty list?

Also, this can be rewritten as `old_value = values or old_value`.

Reply:
I think I updated this code again and it slipped my mind that the values itself can be a empty array as it is already wrapped in a `list` method right ?